File: fixtures/cpp/controller-service-provider/tests/payment_flow_test.cpp
#include "payment_service.hpp"

using chakra::payments::PaymentController;
using chakra::payments::PaymentProvider;
using chakra::payments::PaymentService;

void test_refund_flow() {
  PaymentProvider provider;
  PaymentService service(provider);
  PaymentController controller(service);
  controller.handle_refund("payment-42");
  chakra::payments::controller_refund();
}

File: fixtures/typescript/controller-service-provider/tests/paymentFlow.test.ts
import { PaymentController } from "../src/api/controller";
import { PaymentService } from "../src/service/paymentService";

describe("payment flow", () => {
    it("refund delegates to provider", () => {
        const controller = new PaymentController();
        controller.refund(1250);
    });

    it("service refund reaches the shared target", () => {
        const service = new PaymentService();
        service.refund(500);
    });
});
